measurement/reporting.py

```diff
@@ -177,6 +178,7 @@ def save_report_data(data: pd.DataFrame) -> None:
     )
 
 
+@lru_cache
```

Comment:
What is this used for?

Reply:
It's used to cache calls to the decorated function, so if called with exactly the same inputs, the function is not run and just the same result is returned. Useful in this case to avoid querying the DB multiples times to get the same information.

https://docs.python.org/3/library/functools.html#functools.lru_cache